routing/pathfind.go

```diff
@@ -502,29 +507,41 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 		defer tx.Rollback()
 	}
 
-	if len(r.DestCustomRecords) > 0 {
-		// Check if the target has TLV enabled
-
+	// If no destination features are provided, we will load what features
+	// we have for the target node from our graph.
+	features := r.DestFeatures
+	if features == nil {
 		targetKey, err := btcec.ParsePubKey(target[:], btcec.S256())
 		if err != nil {
 			return nil, err
 		}
 
 		targetNode, err := g.graph.FetchLightningNode(targetKey)
-		if err != nil {
+		switch {
+
+		// If the node exists and has features, use them directly.
+		case err == nil && targetNode.Features != nil:
+			features = targetNode.Features
```

Comment:
Also for this, I thought this was done already in `finalHopParameters`?

Reply:
atm that's done when creating a route, this is before pathfinding is attempted

Reply:
Yes, I mean instantiating `finalHopParameters` before pathfinding already, so that this logic doesnt need to be executed twice.

Or otherwise return the path from pathfinding so that it includes everything `newRoute` needs. Something doesn't feel right in the split between pathfinding and `newRoute`.

Reply:
fixed by moving checks to pathfinding, and only having lighter checks in route construction